controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -384,6 +386,47 @@ func (in *KubeadmControlPlane) validateEtcd(prev *KubeadmControlPlane) (allErrs
 	return allErrs
 }
 
+func (in *KubeadmControlPlane) validateVersion(previousVersion string) (allErrs field.ErrorList) {
+	fromVersion, err := util.ParseMajorMinorPatch(previousVersion)
+	if err != nil {
+		allErrs = append(allErrs,
+			field.InternalError(
+				field.NewPath("spec", "version"),
+				errors.Wrapf(err, "failed to parse current kubeadmcontrolplane version: %s", previousVersion),
+			),
+		)
+		return allErrs
+	}
+
+	toVersion, err := util.ParseMajorMinorPatch(in.Spec.Version)
+	if err != nil {
+		allErrs = append(allErrs,
+			field.InternalError(
+				field.NewPath("spec", "version"),
+				errors.Wrapf(err, "failed to parse current kubeadmcontrolplane version: %s", previousVersion),
+			),
+		)
+		return allErrs
+	}
+
+	// since upgrades to the next minor version are allowed, irrespective of the patch version
+	ceilVersion := semver.Version{
+		Major: fromVersion.Major,
+		Minor: fromVersion.Minor + 2,
+		Patch: 0,
+	}
+	if toVersion.GTE(ceilVersion) {
+		allErrs = append(allErrs,
+			field.Forbidden(
+				field.NewPath("spec", "version"),
+				fmt.Sprintf("cannot skip minor versions of kubeadmcontrolplane from %s to %s", previousVersion, in.Spec.Version),
```

Comment:
```suggestion
				fmt.Sprintf("cannot update Kubernetes version from %s to %s", previousVersion, in.Spec.Version),
```

We should probably add some more context here, do we have a link handy to Kubernetes version upgrade policy? 

cc @fabriziopandini